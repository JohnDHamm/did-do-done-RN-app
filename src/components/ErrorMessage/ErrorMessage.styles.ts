import styled from 'styled-components/native';
import { COLORS } from '../../styles';

export const ErrorMsg = styled.Text`
  color: ${COLORS.MISSED_RED};
  font-size: 14px;
`;
