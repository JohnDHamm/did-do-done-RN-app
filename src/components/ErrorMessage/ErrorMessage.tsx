import React from 'react';
import { ErrorMsg } from './ErrorMessage.styles';

const ErrorMessage: React.FC = ({ children }) => {
  return <ErrorMsg>{children}</ErrorMsg>;
};

export default ErrorMessage;
