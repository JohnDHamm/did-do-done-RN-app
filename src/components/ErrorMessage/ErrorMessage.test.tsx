import React from 'react';
import { cleanup, render } from 'react-native-testing-library';
import ErrorMessage from './ErrorMessage';

describe('ErrorMessage', () => {
  afterEach(cleanup);

  it('renders correctly', () => {
    const { toJSON } = render(<ErrorMessage>error message</ErrorMessage>);

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders the text correctly', () => {
    const { getByText } = render(<ErrorMessage>error message</ErrorMessage>);

    expect(getByText('error message')).not.toBeNull;
  });
});
