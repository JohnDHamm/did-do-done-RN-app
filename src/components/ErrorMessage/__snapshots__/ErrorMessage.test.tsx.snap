// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#B70000",
        "fontSize": 14,
      },
    ]
  }
>
  error message
</Text>
`;
